core/trino-main/src/test/java/io/trino/sql/TestLikeFunctions.java

```diff
@@ -69,8 +80,55 @@ public void testLikeChar()
+
+        // pattern of length equal to value length
+        assertFunction("CAST('foo' AS char(3)) LIKE 'foo'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(3)) LIKE 'fob'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo   '", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo __'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE '%%%%%%'", BOOLEAN, true);
+
+        // pattern longer than value length
+        assertFunction("CAST('foo' AS char(3)) LIKE '%%foo'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(3)) LIKE 'f\\_\\_' ESCAPE '\\'", BOOLEAN, false);
+        assertFunction("CAST('f__' AS char(3)) LIKE 'f\\_\\_' ESCAPE '\\'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo    '", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo __ '", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE '_______'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE '%%%%%%%%%%%%'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo   %%%%%%%%%%%%'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo  %%%%%%%%%%%% '", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo  %%%%%%%%%%%%  '", BOOLEAN, false);
+        assertFunction("CAST('foobar' AS char(6)) LIKE 'foobar%%%%%%%%%%%%'", BOOLEAN, true);
+
+        // pattern of char type
+        assertFunction("'foo' LIKE CAST('foo' AS char(4))", BOOLEAN, false);
+        assertFunction("'foo' LIKE CAST('fo_' AS char(4))", BOOLEAN, false);
+        assertFunction("'foo' LIKE CAST('___' AS char(4))", BOOLEAN, false);
+        assertFunction("'foo' LIKE CAST('foo%' AS char(4))", BOOLEAN, true);
+        assertFunction("'foo' LIKE CAST('foo_' AS char(4))", BOOLEAN, false);
+        assertFunction("'foo ' LIKE CAST('foo' AS char(3))", BOOLEAN, false);
+        assertFunction("'foo ' LIKE CAST('fo_' AS char(3))", BOOLEAN, false);
+        assertFunction("'foo ' LIKE CAST('fo%' AS char(3))", BOOLEAN, true);
+
+        // value and patter of char type
+        assertFunction("CAST('foo' AS char(3)) LIKE CAST('foo' AS char(4))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(3)) LIKE CAST('fo_' AS char(4))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(3)) LIKE CAST('___' AS char(4))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(3)) LIKE CAST('foo%' AS char(4))", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(3)) LIKE CAST('foo_' AS char(4))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(4)) LIKE CAST('foo' AS char(3))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(4)) LIKE CAST('fo_' AS char(3))", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(4)) LIKE CAST('fo%' AS char(3))", BOOLEAN, true);
```

Comment:
Add tests with truncation on either side of the comparison.

Reply:
i think we don't need to test the truncation here, it's orthogonal, i.e. someone else's responsibility